### ISSUE TITLE
feat(rules): enforce ARIA naming prohibition on elements without explicit role

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3630]
## Summary

The `wai-aria` rule should detect `aria-label`, `aria-labelledby`, and `aria-braillelabel` on elements with "naming prohibition" when no explicit `role` is set.

## What to enforce

Per ARIA in HTML §3, certain elements (e.g., `<span>`, `<div>`, `<p>`, `<b>`, `<code>`, `<em>`, etc.) have a "Naming Prohibition" in their default (implicit) role. These elements must not use `aria-label`, `aria-labelledby`, or `aria-braillelabel` unless an explicit `role` that supports naming is specified.

## Example

```html
<!-- Invalid: span has naming prohibition with its implicit role -->
<span aria-label="hello">text</span>

<!-- Valid: explicit role that supports naming -->
<span role="button" aria-label="hello">text</span>
```

## Spec reference

- https://w3c.github.io/html-aria/ (§3, per-element tables)

## Impact

~114 missed errors in the nu-html-checker compatibility benchmark.

**Comment by YusukeHirao** [comment]
### Benchmark update (58.3%)

ARIA naming prohibition (114 files) remains in the missed errors under "Must-not constraint" category. Total Must-not constraints: 100 missed.

Still requires wai-aria rule modification or new rule. May benefit from #3645 (wai-aria rule split).

**Comment by YusukeHirao** [comment]
v5.0 推奨（C群）として #3684 に整理。missed error ~114件。v5.x マイナーでも追加可。
